### PR TITLE
forward error to reply.send() instead of done() onRequest and preHandler

### PR DESCRIPTION
**Pull request by jackneer**
update document based on #2561 forward error to reply.send() instead of done() onRequest and preHandler from adding a hook. Please review, Thanks!

### CODE REVIEW

**Comment by mcollina** [comment]
This piece is correct, you should update the lifecycle diagram.

**Comment by jackneer** [comment]
@mcollina I updated lifecycle diagram. do u mean to remove my first commit by 'This piece is correct' ?

**Review by mcollina — APPROVED** [review]
lgtm

File: docs/Hooks.md [hunk 1]
```diff
@@ -227,19 +227,19 @@ fastify.addHook('onTimeout', async (request, reply) => {
...
+  reply.send(new Error('Some error'))
 })
 ```
 
 If you want to pass a custom error code to the user, just use `reply.code()`:
 ```js
 fastify.addHook('preHandler', (request, reply, done) => {
   reply.code(400)
-  done(new Error('Some error'))
+  reply.send(new Error('Some error'))
```
Comment (mcollina) [inline docs/Hooks.md:242]: These needs to be restored

**Review by mcollina — CHANGES_REQUESTED** [review]

**Comment by mcollina** [comment]
> do u mean to remove my first commit by 'This piece is correct' ?

yes remove the first commit

**Comment by jackneer** [comment]
reverted 1st commit, please review, thanks!

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**